modules/vnet_peering/main.tf

```
@@ -0,0 +1,31 @@
+data "azurerm_virtual_network" "local_peer" {
+  name                = var.local_peer_config.vnet_name
+  resource_group_name = var.local_peer_config.resource_group_name
+}
+
+data "azurerm_virtual_network" "remote_peer" {
+  name                = var.remote_peer_config.vnet_name
+  resource_group_name = var.remote_peer_config.resource_group_name
+}
+
+resource "azurerm_virtual_network_peering" "local" {
+  name                         = try(var.local_peer_config.name, "${var.name_prefix}${var.local_peer_config.vnet_name}-to-${var.remote_peer_config.vnet_name}")
```

```suggestion
  name                         = try(var.local_peer_config.name, "${var.local_peer_config.vnet_name}-to-${var.remote_peer_config.vnet_name}")
```

two reasons for that change:
- when working with existing resource name (like `vnet_name`, it has to be a full name of an existing resource, with prefixes and everything, as we use it in a datasource) we would already have a prefix starting the peering name, so for `hub-` prefix this would give us: `hub-hub-vnet-hub-othervnet` <- does not make sense
- this is not a resource, meaning you cannot see it in a resource group (if I'm not wrong here), hence it's name should be descriptive, but does not need to have a prefix IMO, you will see it inside a VNET in the peerings list